### PR TITLE
Sscs 10548 panel members are excluded when set aside granted

### PR DESCRIPTION
**Pull request by amoslv1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10548

### Change description ###
Implements changes from SSCS-10865 and SSCS-10544
***No additions besides those two branches***

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SchedulingAndListingFields.java [hunk 1]
```diff
@@ -19,7 +22,24 @@ public class SchedulingAndListingFields {
...
+    private PanelMemberExclusions panelMemberExclusions;
     private OverrideFields overrideFields;
     private OverrideFields defaultListingValues;
     private List<AmendReason> amendReasons;
+
+    @SuppressWarnings("unused")
+    @JsonIgnore
+    public PanelMemberExclusions getPanelMemberExclusions() {
+        if (panelMemberExclusions == null) {
+            this.panelMemberExclusions = PanelMemberExclusions.builder().build();
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SchedulingAndListingFields.java:35]: why **this**? cant see a param on the function so you dont have to use **this**

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private final JudicialRefDataApi judicialRefDataApi;
+    private final IdamService idamService;
+
+    public String getJudicialUserFullName(String personalCode) {
+        log.info("Requesting Judicial User with personal code {}", personalCode);
+        IdamTokens idamTokens = idamService.getIdamTokens();
+
+        List<JudicialUser> judicialUsers = judicialRefDataApi.getJudicialUsers(idamTokens.getIdamOauth2Token(),
+            idamTokens.getServiceAuthorization(), JudicialRefDataUsersRequest.builder()
+                .personalCodes(List.of(personalCode)).build());
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:28]:
```
JudicialRefDataUsersRequest.builder()
                .personalCodes(List.of(personalCode)).build()
```
as separate var please

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class JudicialRefDataService {
+
+    private final JudicialRefDataApi judicialRefDataApi;
+    private final IdamService idamService;
+
+    public String getJudicialUserFullName(String personalCode) {
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:22]: Do we have any unit tests covering this?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/PanelMemberExclusions.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import uk.gov.hmcts.reform.sscs.model.client.JudicialUserBase;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Builder(toBuilder = true)
+@AllArgsConstructor
+@NoArgsConstructor
+@JsonInclude
+public class PanelMemberExclusions {
+    private YesNo arePanelMembersExcluded;
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/PanelMemberExclusions.java:20]: Extremely minor. Could this be reordered to have the YesNo values next to one another? Fine if not changed.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Requesting Judicial User with personal code {}", personalCode);
+        IdamTokens idamTokens = idamService.getIdamTokens();
+
+        List<JudicialUser> judicialUsers = judicialRefDataApi.getJudicialUsers(idamTokens.getIdamOauth2Token(),
+            idamTokens.getServiceAuthorization(), JudicialRefDataUsersRequest.builder()
+                .personalCodes(List.of(personalCode)).build());
+
+        return judicialUsers.get(0).getFullName();
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:30]: Are we always expecting the first list item to be the one we return here?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java (hunk 2 above)
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:28]: Avoid nested builders.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java (hunk 5 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:30]: yes. We are sending the personal code which is unique to each user so will only get one back

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java (hunk 2 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:28]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java (hunk 3 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:22]: We have a functional test that hits this in tribs but not unit, created a unit test for it now in common

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/PanelMemberExclusions.java (hunk 4 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/PanelMemberExclusions.java:20]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java (hunk 2 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java:28]: done

**Review by jarekPierchala — APPROVED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Pull request closed without merging**